force_wfmanager/ui/setup/workflow_tree.py

```diff
@@ -646,7 +646,37 @@ def new_layer(self, ui_info, object):
     @triggers_verify
     def new_mco(self, ui_info, object):
         """Adds a new mco to the workflow"""
-        object.model.mco_model = self.system_state.entity_creator.model
+
+        # Try and set the new MCO model's parameters and kpis to those
+        # of the old. Afterall the user might want to change the optimizer,
+        # but keep the same parameters and kpis.
+        try:
+
+            # Get the old model's parameters and kpis.
+            # AttributeError thrown if these do not exist.
+            params = object.model.mco_model.parameters
+            kpis = object.model.mco_model.kpis
+
+            # change to the new model
+            object.model.mco_model = self.system_state.entity_creator.model
+
+            # Set the new model's parameters and kpis to that of the old.
+            # Each parameter's factory attribute should refer to a new
+            # instance of the relevant MCOParameterFactory, that references
+            # the appropriate MCOFactory, plugin-id and plugin-name.
+            # This is done by calling MCOParameterFactory(<new MCOFactory>).
+            params_new = []
+            for p in params:
+                p.factory = p.factory.__class__(object.model.mco_model.factory)
```

Comment:
Note: We should try not refer to protected attributes of a class externally - if there is a missing functionality then it's best to introduce this into the class itself and provide a public method to call. 

Additionally, I think we need to check here whether the each of the old `BaseMCOParameterFactory` classes are supported by the new `BaseMCOFactory` class (`object.model.mco_model.factory`). 

Putting both these together, we could use the `BaseMCOFactory.parameter_factory_classes` attribute to have have something like:

```python
mco_factory = object.model.mco_model.factory

for p in params:
    for factory_cls in mco_factory.parameter_factory_classes:
        if isinstance(p.factory, factory_cls):
            p.factory = factory_cls(mco_factory)
            params_new.append(p)
```